### PR TITLE
improved yarn berry setup for a new app creation

### PR DESCRIPTION
**Pull request by sio4**
Please take a look at this patch PR for gobuffalo/cli/pull/96.

### CODE REVIEW

File: internal/genny/assets/webpack/webpack.go [hunk 1]
```diff
@@ -13,6 +13,8 @@ import (
 	"github.com/gobuffalo/genny/v2/gogen"
 )
 
+const YARN = "yarnpkg" // CHKME: could it be changed to yarn?
```
Comment (saurori) [inline internal/genny/assets/webpack/webpack.go:16]: Probably can just use `yarn`. `yarnpkg` is an alias pointing to `bin/yarn.js`

File: internal/genny/assets/webpack/webpack.go [hunk 2]
```diff
@@ -123,9 +126,31 @@ func (y yarnWriter) Write(p []byte) (int, error) {
...
+	if err := r.Exec(exec.Command(YARN, "set", "version", "berry")); err != nil {
+		return err
+	}
+	// the .yarnrc.yml format is different between classic and berry so
+	// it should be configured after berry installation (set version)
+	if err := r.Exec(exec.Command(YARN, "config", "set", "enableGlobalCache", "true")); err != nil {
+		return err
+	}
+	// CHKME: can we enable it only for testing? or unset after installation?
+	if err := r.Exec(exec.Command(YARN, "config", "set", "enableImmutableInstalls", "false")); err != nil {
```
Comment (saurori) [inline internal/genny/assets/webpack/webpack.go:149]: Do you know if these `yarn config set` commands are global or project based? If they are project based, then it seems the `.yarnrc.yml` template will get overridden.

I think the better option is to set the environment variable for CI:
>The enableImmutableInstalls will now default to true on CI (we still recommend to explicitly use --immutable on the CLI).
You can re-allow mutations by adding YARN_ENABLE_IMMUTABLE_INSTALLS=false in your environment variables.
https://github.com/yarnpkg/berry/blob/master/CHANGELOG.md#300

It looks like this is what yarn's own [end-to-end CI does](https://github.com/yarnpkg/berry/blob/4033a1b8f70e0de3b5bee0ba3fe595aed7c25e88/scripts/e2e-setup-ci.sh#L26).

**Review by saurori — COMMENTED** [review]
I think we should explore using the env var `YARN_ENABLE_IMMUTABLE_INSTALLS=0` for CI

**Comment by sio4** [comment]
Feel free to fix those parts if you wish. I just wanted to give you a patch I wrote after I dig the issue. I hope your PR could be merged soon :-)

**Comment by saurori** [comment]
Closing. Implemented most of this on https://github.com/gobuffalo/cli/pull/96

**Pull request closed without merging**